src/reynir/verbframe.py

```diff
@@ -248,86 +255,162 @@ def matches_pcl(self, particle: str) -> bool:
+        def get_score(s: str) -> Tuple[str, Optional[int]]:
+            # Start by handling the $score() pragma, if present
+            score: Optional[int] = None
+            ix = s.rfind("$score(")  # Must be at the end
+            if ix >= 0:
+                sc = s[ix:]
+                s = s[0:ix].strip()
+                if not sc.endswith(")"):
+                    raise ConfigError("Invalid score pragma; form should be $score(n)")
+                # There is an associated score with this verb form, to be taken
+                # into consideration by the reducer
+                sc = sc[7:-1].strip()
+                try:
+                    score = int(sc)
+                except ValueError:
+                    raise ConfigError("Invalid score ('{0}') for verb form".format(sc))
+            return s, score
+
+        def get_error(s: str) -> Tuple[str, Optional[str]]:
+            # Check for $error
+            error = None
+            ix = s.rfind("$error(")
+            if ix >= 0:
+                if not s.endswith(")"):
+                    raise ConfigError("Invalid error pragma; form should be $error(...)")
+                error = s[ix + 7 : -1].strip()
+                s = s[0:ix].strip()
+                if not error:
+                    raise ConfigError("Expected error specification in $error(...)")
+            return s, error            
+
+        def get_particle(s: str) -> Tuple[str, Optional[str]]:
+            # Process particles, should only be one in each line
+            particle = None
+            ix = s.rfind("*")
+            if ix >= 0:
+                particle = s[ix:].strip()
+                s = s[0:ix].strip()
+                if " " in particle:
+                    raise ConfigError("Particle should only be one word")
+                elif len(particle) < 2:
+                    raise ConfigError("Particle should be at least one letter")
+            return s, particle
+
+        def get_preposition(s: str) -> Tuple[str, List[Tuple[str, str]]]:
+            # Process preposition arguments, if any
+            prepositions: List[Tuple[str, str]] = []
+            ap = s.split("/")
+            s = ap[0]
+            ix = 1
+            while ix < len(ap):
+                # We expect something like 'af þgf', or possibly
+                # 'fyrir_hönd þf' (where the underscore needs to be replaced by a space)
+                # or 'milli ef_ft' (detailing the plural)
+                p = ap[ix].strip()
+                parg = p.split()
+                #if len(parg) != 2:
+                #    raise ConfigError("Preposition should have exactly one argument")
+                if parg[1] not in ALL_CASES and parg[1] not in SUBCLAUSES:
+                    parg[1] = REFLPRN.get(parg[1], parg[1])
+                    assert parg[1] is not None
+                    spl = parg[1].split("_")
+                    skip = ["gr", "ft", "est", "mst", "et", "kk", "kvk", "hk"]   # TODO Handle these variants
```

Comment:
Convert to `frozenset()` and move to constant at beginning of file

Reply:
Done